### ISSUE TITLE
False positives about unused variable and variable initialization

### ISSUE THREAD

**Issue by zedxxx** [integrated-application-development/delphilint#49]
### Prerequisites

- [X] This bug is in DelphiLint, not SonarQube or the Delphi IDE.
- [X] This bug has not already been reported.
- [X] This bug is not a problem related to analysis quality (e.g. parsing errors, false positive issues), which should be raised on the [SonarDelphi repository](https://github.com/integrated-application-development/sonar-delphi/issues).

### Delphi IDE version

Delphi 12.1

### DelphiLint version

1.1.0

### SonarDelphi version

1.6.0

### Issue description

```delphi
unit Unit1;

interface

uses
  GR32,
  GR32_Polygons,
  GR32_VectorUtils;

procedure DrawThickLine(ABitmap: TBitmap32; const AStart, AEnd: TFloatPoint; const AColor: TColor32;
  const AWidth: Integer); inline;

procedure DrawThickPolyLine(ABitmap: TBitmap32; const APoints: TArrayOfFloatPoint; const AColor: TColor32;
  const AWidth: Integer);

implementation

procedure DrawThickLine(ABitmap: TBitmap32; const AStart, AEnd: TFloatPoint; const AColor: TColor32;
  const AWidth: Integer);
var
  VLine: TArrayOfFloatPoint;
begin
  SetLength(VLine, 2);

  VLine[0] := AStart;
  VLine[1] := AEnd;

  DrawThickPolyLine(ABitmap, VLine, AColor, AWidth);
end;

procedure DrawThickPolyLine(ABitmap: TBitmap32; const APoints: TArrayOfFloatPoint;
  const AColor: TColor32; const AWidth: Integer);
var
  VPoly: TArrayOfFloatPoint;
begin
  VPoly := BuildPolyLine(APoints, AWidth);

  if not ABitmap.MeasuringMode then begin
    PolygonFS(ABitmap, VPoly, AColor);
  end;

  ABitmap.Changed(MakeRect(PolygonBounds(VPoly), rrOutside));
end;

end.
```

### Steps to reproduce

![1](https://github.com/integrated-application-development/delphilint/assets/6298130/8f4230a9-f8ae-44b4-80bc-a1d37feb64eb)

**Comment by zedxxx** [comment]
Sorry, wrong place. Reported here: https://github.com/integrated-application-development/sonar-delphi/issues/250
